server/monitor.go

```diff
@@ -0,0 +1,504 @@
+	HBInbox       string                      `json:"hb_inbox"`
+	Subscriptions map[string][]*Subscriptionz `json:"subscriptions,omitempty"`
+}
+
+// Channelsz lists the name of all NATS Streaming Channelsz
+type Channelsz struct {
+	ClusterID string    `json:"cluster_id"`
+	ServerID  string    `json:"server_id"`
+	Now       time.Time `json:"now"`
+	Offset    int       `json:"offset"`
+	Limit     int       `json:"limit"`
+	Count     int       `json:"count"`
+	Total     int       `json:"total"`
+	Channels  []string  `json:"channels"`
+}
+
+// ChannelsWithSubsz list detailed channels along with subscriptions on each channel
+type ChannelsWithSubsz struct {
+	ClusterID string      `json:"cluster_id"`
+	ServerID  string      `json:"server_id"`
+	Now       time.Time   `json:"now"`
+	Offset    int         `json:"offset"`
+	Limit     int         `json:"limit"`
+	Count     int         `json:"count"`
+	Total     int         `json:"total"`
+	Channels  []*Channelz `json:"channels"`
+}
+
+// Channelz describes a NATS Streaming Channel
+type Channelz struct {
+	Name          string           `json:"name"`
+	Msgs          int              `json:"msgs"`
+	Bytes         uint64           `json:"bytes"`
+	FirstSeq      uint64           `json:"first_seq"`
+	LastSeq       uint64           `json:"last_seq"`
+	Subscriptions []*Subscriptionz `json:"subscriptions,omitempty"`
+}
+
+// Subscriptionz describes a NATS Streaming Subscription
+type Subscriptionz struct {
+	Inbox        string `json:"inbox"`
+	AckInbox     string `json:"ack_inbox"`
+	DurableName  string `json:"durable_name,omitempty"`
+	QueueName    string `json:"queue_name,omitempty"`
+	IsDurable    bool   `json:"is_durable"`
+	MaxInflight  int    `json:"max_inflight"`
+	AckWait      int    `json:"ack_wait"`
+	LastSent     uint64 `json:"last_sent"`
+	PendingCount int    `json:"pending_count"`
+	IsStalled    bool   `json:"is_stalled"`
+}
+
+func (s *StanServer) startMonitoring(nOpts *gnatsd.Options) error {
+	var hh http.Handler
+	// If we are connecting to remote NATS Server, we start our own
+	// HTTP(s) server.
+	if s.opts.NATSServerURL != "" {
+		s.natsServer = gnatsd.New(nOpts)
+		if err := s.natsServer.StartMonitoring(); err != nil {
+			return err
+		}
+		hh = s.natsServer.HTTPHandler()
+	} else {
+		// We may need to wait for the HTTPHandler to be ready.
+		timeout := time.Now().Add(10 * time.Second)
+		for time.Now().Before(timeout) {
+			if hh = s.natsServer.HTTPHandler(); hh != nil {
+				break
+			}
+			time.Sleep(250 * time.Millisecond)
+		}
+	}
+	if hh == nil {
+		return fmt.Errorf("unable to start monitoring server")
+	}
+
+	mux := hh.(*http.ServeMux)
+	mux.HandleFunc(RootPath, s.handleRootz)
+	mux.HandleFunc(ServerPath, s.handleServerz)
+	mux.HandleFunc(StorePath, s.handleStorez)
+	mux.HandleFunc(ClientsPath, s.handleClientsz)
+	mux.HandleFunc(ChannelsPath, s.handleChannelsz)
+
+	return nil
+}
+
+func (s *StanServer) handleRootz(w http.ResponseWriter, r *http.Request) {
```

Comment:
No we can't. The `HandleFunc` would return the error `http: multiple registrations for ...`, this is why I had to have a "sub" root for streaming.

So the idea is that the doc (and banner on startup?) would refer to "http://localhost:port/streaming". One can then manually go to NATS monitoring by removing `streaming`. Note that there is situations where NATS monitoring would report nothing when the Streaming server does not embed the NATS server.